storage/interceptors/broker_notifications_interceptor.go

```diff
@@ -164,19 +158,79 @@ func fetchBrokerPlans(ctx context.Context, brokerID string, repository storage.R
 	return objList.(*types.ServicePlans).ServicePlans, nil
 }
 
-func getSupportedPlatformsForPlans(plans []*types.ServicePlan) []string {
-	platformTypes := make(map[string]bool)
+func ResolveSupportedPlatformIDsForPlans(ctx context.Context, plans []*types.ServicePlan, repository storage.Repository) ([]string, error) {
+	var platformTypes map[string]bool
+	platformNames := make(map[string]bool)
```

Comment:
Why create `map[string]bool` instead of `[]string` for both `platformTypes` and `platformNames`? You don't do anything special with these and you immediately after that convert them to slices anyway.

Reply:
This is a "set" used tto collect them from multiple plans. That was how it was implemented before so i followed the pattern.